gensim/models/sldamodel.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2011 Radim Rehurek <radimrehurek@seznam.cz>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+import logging
+import numpy as np
+import six
+from scipy.special import gammaln, psi  # gamma function utils
+from scipy.special import polygamma
+from six.moves import xrange
+
+from gensim import interfaces, utils, matutils
+from gensim.matutils import dirichlet_expectation
+from gensim.models import basemodel, CoherenceModel
+
+# log(sum(exp(x))) that tries to avoid overflow
+try:
+    # try importing from here if older scipy is installed
+    from scipy.maxentropy import logsumexp
+except ImportError:
+    # maxentropy has been removed in recent releases, logsumexp now in misc
+    from scipy.misc import logsumexp
+
+
+logger = logging.getLogger('gensim.models.sldamodel')
+
+def dirichlet_expectation(alpha):
+    """
+    For a vector theta ~ Dir(alpha), computes E[log(theta)] given alpha.
+    """
+    if (len(alpha.shape) == 1):
+        return(psi(alpha) - psi(n.sum(alpha)))
+    return(psi(alpha) - psi(n.sum(alpha, 1))[:, n.newaxis])
+
+def get_lambda(self):
+        return self.eta + self.sstats
+
+def get_Elogbeta(self):
+        return dirichlet_expectation(self.get_lambda())
+
+class sLdaModel(interfaces.TransformationABC, basemodel.BaseTopicModel):
+
+    def __init__(self, corpus=None, n_topics=100, alpha='symmetric', beta, mu, nu, nu2,
+                 sigma2, iterations = 100, report_iter=10, seed=None):
+    """
+    Supervised latent Dirichlet allocation, using collapsed Gibbs
+    sampling.
+    Args
+        n_topics : int
+            Number of topics
+        alpha : array-like, shape = (n_topics,)
+            Dirichlet distribution parameter for each document's topic
+            distribution.
+        beta : array-like, shape = (n_terms,)
+            Dirichlet distribution parameter for each topic's term distribution.
+        mu : float
+            Mean of regression coefficients (eta).
+        nu2 : float
+            Variance of regression coefficients (eta).
+        sigma2 : float
+            Variance of response (y).
+        n_iter : int, default=100
+            Number of iterations of Gibbs sampler
+        n_report_iter : int, default=10
+            Number of iterations of Gibbs sampler between progress reports.
+        seed : int, optional
```

Comment:
Missed parameters: report_iter, iterations, nu, corpus
New parameters (that not defined in \_\_init\_\_): n_iter, n_report_iter.